libft/srcs/ft_hash/update_value.c

```diff
@@ -1,29 +1,18 @@
 #include <stdlib.h>
 #include "ft_deque.h"
 #include "ft_hash.h"
-#include "ft_string.h"
 
 // key exist in hash
 // free(key, pre-content), update new content
-void	update_content_of_key(t_hash *hash, char *key, void *content)
+void	update_content_of_key(char **key, \
+								void *content, \
+								t_deque_node *target_node)
 {
-	uint64_t		hash_val;
-	t_deque_node	*node;
-	t_elem			*elem;
+	t_elem	*elem;
 
-	hash_val = generate_fnv_hash_64((unsigned char *)key, hash->table_size);
-	node = hash->table[hash_val]->node;
-	while (node)
-	{
-		elem = (t_elem *)node->content;
-		if (ft_streq(elem->key, key))
-		{
-			// free(key);
-			// todo: del func
-			// free(node->content);
-			elem->content = content;
-			return ;
-		}
-		node = node->next;
-	}
+	free(*key);
+	*key = NULL;
+	elem = (t_elem *)target_node->content;
+	free(elem->content);
```

Comment:
```suggestion
	del(elem->content);
```
ここはdelかもですね！
（現状はdelの実態がfreeなので変わらないですけど）

delが渡されないとvoid *contentを適切に削除できないので、
色々な関数にdelを渡さないといけない気がしてきました...